File: src/com/gdb/domain/SalaryAccount.java
package com.gdb.domain;
import com.gdb.exceptions.*;

public class SalaryAccount extends AbstractAccount {
    private String employerName;
    private int inactiveMonths;

    public SalaryAccount(String accountNumber, String name, int age, double balance, String status, String pin, String employerName) {
        super(accountNumber, name, age, balance, "SALARY", status, pin);
        this.employerName = employerName;
        this.inactiveMonths = 0;
    }

    @Override
    protected void processDebit(double amount) throws AccountException {
        if (amount > this.balance) {
            throw new InsufficientBalanceException("Insufficient funds in account");
        }
        this.balance -= amount;
    }
}

File: gdb/domain/salary_account.py

from gdb.domain.abstract_account import AbstractAccount
from gdb.exceptions import InsufficientBalanceException

class SalaryAccount(AbstractAccount):
    def __init__(self, account_number, name, age, balance, status, pin, employer_name):
        super().__init__(account_number, name, age, balance, "SALARY", status, pin)
        
    def process_debit(self, amount: float) -> None:
        self.balance -= amount
